packages/x-components/src/utils/throttle.ts

```diff
@@ -5,27 +5,46 @@ import { ThrottleFunction } from './types';
  *
  * @param fn - Function to be debounced.
  * @param throttleTimeMs - The time of throttle in ms.
+ * @param options - Advanced options to throttle the function with.
  * @returns A new function with the throttle.
  *
  * @public
  */
 
 export const throttle = <Params extends any[]>(
   fn: (...args: Params) => void,
-  throttleTimeMs: number
+  throttleTimeMs: number,
+  { leading = false }: ThrottleOptions = {}
```

Comment:
Just to understand it, Is this change needed for this task?

Reply:
I am in the same spot, i don't know why the change is needed

Reply:
Yep, it is not needed for this task.
Take it as a gift from our beloved @JaviIG 🎁

Reply:
It can be removed as far as I remember. It is not needed, as requests go one by one. Either that approach or a simple delay is enough 
Salu2 ❤️